### ISSUE TITLE
Support 'env' in sampling strategies file for remote sampling

### ISSUE THREAD

**Issue by annanay25** [jaegertracing/jaeger#2024]
## Requirement - what kind of business use case are you trying to solve?

Support different sampling rates for the same endpoints in different environments (dev/staging/prod).

## Problem - what in Jaeger blocks you from solving the requirement?

The sampling strategies config does not have a top level `env` construct. Ideally, we would like to do -
```curl http://jaeger-agent:5778/sampling?env='dev'&service='loki-gateway'``` 

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Extend the config to include `env` construct, and expose it as an API parameter.

## Any open questions to address
-

cc @joe-elliott @cyriltovena

**Comment by cyriltovena** [comment]
Yes please ! I use jaeger every day and to be honest this is difficult to work around when you have one environment with huge throughput, but you still want your dev workload to be 100% sampled.

We may want to make this more generalized and support tags overall.

**Comment by ScottKaye** [comment]
Are you using the same jaeger-agent for dev/staging/prod?

**Comment by joe-elliott** [comment]
No, but we are using the same collectors/backend for dev/staging/prod and AFAIK the jaeger agent just proxies the remote sampling query back to the collectors.

**Comment by joe-elliott** [comment]
@jpkrohling No, that's a good point.  It would not be too difficult to partition our collectors by env and have the agents reporting to only a subset of the collectors.
